pkg/scalers/elasticsearch_scaler.go

```diff
@@ -234,11 +234,10 @@ func getValueFromSearch(body []byte, valueLocation string) (int, error) {
 // GetMetricSpecForScaling returns the MetricSpec for the Horizontal Pod Autoscaler
 func (s *elasticsearchScaler) GetMetricSpecForScaling(context.Context) []v2beta2.MetricSpec {
 	targetValue := resource.NewQuantity(int64(s.metadata.targetValue), resource.DecimalSI)
-	metricName := kedautil.NormalizeString(fmt.Sprintf("elasticsearch-%s", s.metadata.searchTemplateName))
 
 	externalMetric := &v2beta2.ExternalMetricSource{
 		Metric: v2beta2.MetricIdentifier{
-			Name: GenerateMetricNameWithIndex(s.metadata.targetValue, metricName),
+			Name: GenerateMetricNameWithIndex(s.metadata.targetValue, s.metadata.metricName),
```

Comment:
You should call only once to `GenerateMetricNameWithIndex`. This index is not related with any inside the scaler, it's related with the trigger index inside the ScaledObject to avoid conflicts with duplicated names in the same ScaledObject.
You could call to the function here or in the constructor like you are doing but don't do it both, because the final metric name with current code will be like `s{s.metadata.targetValue}-s{config.ScalerIndex}-elasticsearch-%s`. It should be like  `s{config.ScalerIndex}-elasticsearch-%s`